src/document.jl

```diff
@@ -81,18 +81,17 @@ TokenDocument(txt::AbstractString) = TokenDocument(String(txt), DocumentMetadata
 
 mutable struct NGramDocument{T<:AbstractString} <: AbstractDocument
     ngrams::Dict{T,Int}
-    n::Int
+    n::Union{Int,Vector{Int}}
```

Comment:
Would this defeat type inference?

Reply:
I don't seen `ngram_complexity` (or this field) used anywhere in TextAnalysis. It may be used from external code. But then there are no abstracts here, and it should be easy to make them type stable if/where needed.